src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -828,10 +836,16 @@ def create_arm_client_factory(credentials):
 
     def find_from_user_account(self, username, password, tenant, resource):
         context = self._create_auth_context(tenant)
-        if password:
-            token_entry = context.acquire_token_with_username_password(resource, username, password, _CLIENT_ID)
-        else:  # when refresh account, we will leverage local cached tokens
-            token_entry = context.acquire_token(resource, username, _CLIENT_ID)
+        try:
+            if password:
+                token_entry = context.acquire_token_with_username_password(resource, username, password, _CLIENT_ID)
+            else:  # when refresh account, we will leverage local cached tokens
+                token_entry = context.acquire_token(resource, username, _CLIENT_ID)
+        except InvalidURL as err:
```

Comment:
In some cases from the telemetry, CX tried acquiring token from strange authority url. I don't know how it happens, maybe they use private cloud misconfigured. I just add catch logic here.

Reply:
Users can register their own cloud with `az cloud register` and provide invalid URL for AAD. Consider mark that as a `UserFault`.

Reply:
I actually use `UnknownError`, which is a subclass of `UserFault`